deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/PackageManagerInstall.java

```diff
@@ -54,12 +56,32 @@ public static Installation install(PackageManagerInstallConfig config, final Pro
                 }
             }
 
+            String executionPath = NPM_PATH;
```

Comment:
Should I do the same if `!"provided"` check here like I do for Yarn and PNPM?  Basically only set this download if its something other than "provided"?

Reply:
The "provided" is a bit different, it comes from the mvn frontend plugins. The idea is to use the npm provided by nodejs by default (which is AFAIK not the case for the others).

So here is how I would do it:
- Keep the `provided` by default for the npmVersion
- Let `pnpmVersion` and `yarnVersion` be `Optional`
- Use npm if `npmVersion` is different from `provided` or if no other version is set (then it will use the version provided by nodejs)
- Use yarn if yarnVersion is set (and npm is provided)
- Use pnpm f pnpmVersion is set (and npm is provided and yarnVersion is not set)